appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/model/util/ModelUtils.java

```diff
@@ -416,15 +391,17 @@ public static boolean isAnnotationNull(Annotation annotation) {
                         } else if (String.class.isAssignableFrom(value.getClass()) && !value.toString().isEmpty()) {
                             return false;
                         } else if (value instanceof Annotation) {
-                            allNull = isAnnotationNull((Annotation) value);
+                            if (!isAnnotationNull((Annotation) value)) {
+                                return false;
```

Comment:
FYI: This was a logical slip making an annotation only "empty" if all its fields were "empty" but logically it should be "not all" or "any non empty" makes it "non empty".

Reply:
I'm not quite sure I follow. What's the difference between empty == all fields empty and not empty == at least one non-empty field?

Reply:
The logic that was in place did assign to the `allNull` variable for every property that had an annotation type. So if an annotation would have multiple properties of such types the last of them would effectively make the result of them "all" being null/empty which is wrong logically. The correct logic is all of them should be empty for an annotation to be considered empty which is same as any being non-empty means its non empty.

Reply:
Oh I see, that sounds like a mistake indeed!